unlock-app/src/components/interface/user-account/PaymentDetails.tsx

```diff
@@ -1,217 +1,96 @@
-        this.setState({
-          submitted: false,
-        })
-      }
+    if (result.token) {
+      signPaymentData(result.token.id)
     }
   }
 
-  handleReset = () => {
-    const { errors, close } = this.props
-    errors.forEach(e => close(e))
-    this.handleSubmit()
-  }
-
-  submitButton = () => {
-    const { errors } = this.props
-    const { submitted } = this.state
-
-    if (errors.length) {
-      return (
-        <SubmitButton backgroundColor="var(--red)" onClick={this.handleReset}>
-          Clear Errors and Retry
-        </SubmitButton>
-      )
-    }
-    if (submitted) {
-      return <LoadingButton>Submitting...</LoadingButton>
-    }
-
-    return (
-      <SubmitButton onClick={this.handleSubmit}>
-        Add Payment Method
-      </SubmitButton>
-    )
-  }
-
-  render() {
-    const stripeElementStyles = {
-      base: { fontSize: '16px', lineHeight: '40px' },
-    }
-    return (
-      <Grid>
-        <SectionHeader>Card Details</SectionHeader>
-        <div>
-          <ItemLabel>Cardholder Name</ItemLabel>
-          <Input
-            name="cardHolderName"
-            id="cardHolderName"
-            type="text"
-            placeholder="Cardholder Name"
-            onChange={this.handleInputChange}
-          />
-        </div>
-        <CardContainer>
-          <div>
-            <ItemLabel>Country</ItemLabel>
-            <Input
-              name="addressCountry"
-              id="addressCountry"
-              type="country"
-              placeholder="Country"
-              onChange={this.handleInputChange}
-            />
-          </div>
-          <div>
-            <ItemLabel>Zip / Postal Code</ItemLabel>
-            <Input
-              name="addressZip"
-              id="addressZip"
-              type="text"
-              placeholder="Zip Code"
-              onChange={this.handleInputChange}
-            />
-          </div>
-        </CardContainer>
-        <Column size="full">
-          <Item title="Credit Card Number">
-            <CardNumberElement style={stripeElementStyles} />
-          </Item>
-          <CardContainer>
-            <div>
-              <ItemLabel>Expiry Date</ItemLabel>
-              <CardExpiryElement style={stripeElementStyles} />
-            </div>
-            <div>
-              <ItemLabel>CVC Number</ItemLabel>
-              <CardCVCElement style={stripeElementStyles} />
-            </div>
-          </CardContainer>
-        </Column>
-        <Column size="half">{this.submitButton()}</Column>
-      </Grid>
-    )
+  console.log({
+    stripe,
+    elements
+  })
+
+  const stripeElementOptions = {
+    style: {
+      base: {
+        fontFamily: 'IBM Plex Sans, sans-serif',
+        fontSize: '16px',
+        lineHeight: '60px',
+      },
+    },
   }
+  
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <Input type="text" placeholder="Cardholder Name" name="Cardholder Name" ref={register({required: true})} />
```

Comment:
The `CardElement` actually collects all the important information here, including zip. We still gather the cardholder name, even though it's optional.